slotserver/slot_repository.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) 2020 Hugh Wade
+# SPDX-License-Identifier: MIT
+
+import abc
+import threading
+import pickle
+import os.path
+
+
+class SlotRepositoryInterface(metaclass=abc.ABCMeta):
+    '''
+    Slot server repository.
+    Logical datashape is a Dictionary of Dictionaries.
+    Persists instance data used by IOT dashboard elements.
+    '''
+    @classmethod
+    def __subclasshook__(cls, subclass):
+        return (hasattr(subclass, 'get') and
+                callable(subclass.get) and
+                hasattr(subclass, 'upsert') and
+                callable(subclass.upsert) and
+                hasattr(subclass, 'delete') and
+                callable(subclass.delete) and
+                hasattr(subclass, 'slots') and
+                callable(subclass.slots) and
+                hasattr(subclass, 'subslots') and
+                callable(subclass.subslots) or
+                NotImplemented)
+
+    @abc.abstractmethod
+    def get(self, slot_id: str, subslot_id: str, raise_error: bool) -> object:
+        '''
+        Get the data for a slot.
+        A SlotNotFoundException will be raised if the slot is missing
+        and raise_error is True
+        '''
+        raise NotImplementedError
```

Comment:
How can I cover these raise statements? 
Maybe they can just become pass statements. We have another test that verifies the interface is indeed abstract and can't be constructed